### ISSUE TITLE
TFRecordWriter.flush() do not seem to flush

### ISSUE THREAD

**Issue by cih9088** [tensorflow/tensorflow#26683]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04.5 LTS
TensorFlow installed from (source or binary): pip
TensorFlow version (use command below): v1.12.0-0-ga6d8ffae09 1.12.0
Python version: 3.5.2:
CUDA/cuDNN version: cuda-9.0, cudnn-7.3.1
GPU model and memory: Titan Xp, Titan V

**Describe the current behavior**
```
import tensorflow as tf

mystring = 'start blahblahblah end\n' * 10000

# https://www.zlib.net/manual.html
Z_NO_FLUSH = 0
Z_PARTIAL_FLUSH = 1
Z_SYNC_FLUSH = 2
Z_FULL_FLUSH = 3

writer = tf.python_io.TFRecordWriter('data', tf.io.TFRecordOptions(flush_mode=Z_NO_FLUSH))
writer.write(mystring.encode())
# << - at this point, mystring is not fully written on disk as expected.
writer.flush()
# << - still not fully written
writer.close()
# << - now it is fully written
```

I found `flush_mode` option in `tf.io.TFRecordOptions`.
Although it is only effect `compression_type` is None but anyway I tried from `0` (Z_NO_FLUSH) to `3` (Z_FULL_FLUSH) as stated in https://www.zlib.net/manual.html, none of them work.
Do I miss something or is it a bug?

**Comment by annarev** [comment]
@jsimsa Can you take a look or reassign to someone who knows about record writing?

**Comment by jsimsa** [comment]
@cih9088 the `flush_mode` option is only meaningful when the compression type is ZLIB. For other compression types (e.g. `None` used by your example) it will have no effect.

**Comment by cih9088** [comment]
Is this a bug or intended? https://github.com/tensorflow/tensorflow/pull/27219 seems to try to fix the issue.

**Comment by rmothukuru** [comment]
@cih9088,
Can you please let us know if [this PR](https://github.com/tensorflow/tensorflow/pull/27219) has resolved this issue? Thanks!

**Comment by cih9088** [comment]
It was resolved by #27219 right after the merge but I'm not sure if this is still a problem after all these years because I haven't used it for a long time.

**Comment by rmothukuru** [comment]
Closing this issue as it has been resolved with #27219. Please feel free to open it if the issue still persists. Thanks!
